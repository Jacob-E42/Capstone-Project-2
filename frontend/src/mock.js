import React, { useCallback, useState } from "react";
import UserContext from "./context_providers/UserContext";
import AlertContext from "./context_providers/AlertContext";
import Request from "./api";

const demoUser = {
	id: 11,
	first_name: "testfirst",
	last_name: "testlast",
	email: "test@test.net",
	interests: ["baking", "jetskiing"]
};

const UserProvider = ({ children, user = demoUser }) => {
	const [token, setToken] = useState(null);

	return <UserContext.Provider value={{ user }}>{children}</UserContext.Provider>;
};

const AnonUserProvider = ({ children }) => {
	const [currentUser, setCurrentUser] = useState(null);

	const login = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.login(formData);
				if (token) {
					setCurrentUser(demoUser);
					return { success: true };
				} else throw new Error("Log in token in missing");
			} catch (errors) {
				console.error("login failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	const signup = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.signup(formData);
				console.log(token);
				if (token) setCurrentUser(demoUser);
				else throw new Error("missing sign in token");
				return { success: true };
			} catch (errors) {
				console.error("signup failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	return <UserContext.Provider value={{ currentUser, login, signup }}>{children}</UserContext.Provider>;
};

const AlertProvider = ({ children }) => {
	const [msg, setMsg] = useState("");
	const [color, setColor] = useState("primary");

	return <AlertContext.Provider value={{ msg, setMsg, color, setColor }}>{children}</AlertContext.Provider>;
};

export { UserProvider, AnonUserProvider, AlertProvider };
