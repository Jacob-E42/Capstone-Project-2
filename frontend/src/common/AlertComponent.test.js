import React from "react";
import { render } from "@testing-library/react";
import AlertComponent from "./Alert";
import { AlertProvider } from "../mock";

test("Alert renders without crashing", () => {
	render(
		<AlertProvider>
			<AlertComponent />
		</AlertProvider>
	);
});

test("Alert matches snapshot for success", () => {
	const { asFragment } = render(
		<AlertProvider>
			<AlertComponent msg={"I did it!"} />
		</AlertProvider>
	);
	expect(asFragment()).toMatchSnapshot();
});

test("Alert matches snapshot for danger", () => {
	const { asFragment } = render(
		<AlertProvider>
			<AlertComponent
				color="danger"
				msg="You did not enter the correct information"
			/>
		</AlertProvider>
	);
	expect(asFragment()).toMatchSnapshot();
});
