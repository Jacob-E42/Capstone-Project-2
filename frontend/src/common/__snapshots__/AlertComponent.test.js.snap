// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger alert-dismissible fade"
    role="alert"
  >
    <button
      aria-label="Close"
      class="btn-close"
      type="button"
    />
    You did not enter the correct information
  </div>
</DocumentFragment>
`;

exports[`Alert matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success alert-dismissible fade"
    role="alert"
  >
    <button
      aria-label="Close"
      class="btn-close"
      type="button"
    />
    I did it!
  </div>
</DocumentFragment>
`;
